src/osm_parser.rs

```diff
@@ -317,287 +307,3 @@ pub fn get_priority(element: &ProcessedElement) -> usize {
     // Return a default priority if none of the tags match
     PRIORITY_ORDER.len()
 }
-
-/// Clips a way to the bounding box boundaries using Sutherland-Hodgman algorithm for polygons
-/// or simple line clipping for polylines
-fn clip_way_to_bbox(
```

Comment:
Bug #1 is in here somewhere - we end up chopping things off incorrectly. I've removed this code as it seems like a lot of logic that shouldn't actually be doing much (generally our element processing clamps things to within bounds anyway, right?). But I also understand someone spent a lot of time writing this in the first place. I just don't personally want to be the one to figure out what is wrong in here (plus, there may be more bugs lurking here)